---
title: "Using-Motif-Extraction-mock-DB"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Using-Motif-Extraction-mock-DB}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(UVMProteomics)
```

> This document will only work if NCBI Blast+ is installed in the user PATH.

> If NCBI Blast+ is not in your PATH run this with your path to ncbi-blast-2.14.0+/bin:
```{r, include=F}
# Get the current PATH variable
current_path <- Sys.getenv("PATH")
# Add a new directory to the PATH
new_directory <- "PATH/TO/YOUR/ncbi-blast-2.14.0+/bin/GOES/HERE"
updated_path <- paste(new_directory, current_path, sep = ":")

# Set the updated PATH variable in the R session
Sys.setenv(PATH = updated_path)
```

```{r, include=F}
knitr::opts_chunk$set(
  eval = has_blast_plus_binaries()
)
```

```{r}
# Create a temp output directory
output_directory_path_top <- tempdir()
output_directory_path <- file.path(output_directory_path_top, 'Motif_Extraction_Test')
dir.create(output_directory_path, showWarnings = FALSE)

output_file_path = capture.output(cat(paste0(output_directory_path, "/output.csv")))

# Path to the protein sequence for the reference mouse database 
ref_protein <- file.path(system.file("mock-db/protein.faa", package = "UVMProteomics"))

# Retrieve the NCBI Blast path from PATH variable of environment
path <- Sys.getenv("PATH")
path_directories <- unlist(strsplit(path, ":"))
ncbi_bin <- grep("ncbi-blast-2.14.0+", path_directories, value = TRUE)
#ncbi_bin <- "$PATH/ncbi-blast-2.10.1+/bin/"

#input data sheet
input_path <- file.path(system.file(package = 'UVMProteomics', 'mock-db/mock-db-testFile.xlsx'))

test <- run_motif_extraction(input_path, output_file_path, ncbi_bin, ref_protein)
```
