---
title: "UVM Proteomics"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{UVM Proteomics}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(UVMProteomics)
```

UVMProteomics is a suite of tools for protein related computations and
anaylisis such as motif extraction.

# Installation

UVMProteomics can be installed using the following R code:

```{r devtools installation}
devtools::install_github("jaxlub/UVMProteomics")
```

## Dependencies

### Blast+

NCBI's
[BLAST+](https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/)
suite must be locally installed and accessible in the user's path. NCBI
provides installation instructions for
[Windows](https://www.ncbi.nlm.nih.gov/books/NBK52637/),
[Linux](https://www.ncbi.nlm.nih.gov/books/NBK52640/), and [Mac
OS](https://www.ncbi.nlm.nih.gov/books/NBK569861/). Version 2.14.0+ is
verified compatible with UVMProteomics.

### FASTA Reference File

Not sure how to get?

# Post Translational Modification

## Motif Extraction

This function extracts predetermined protein motifs from a master
protein in a user specified proteome (e.g. [uniProt mouse
proteome](https://www.uniprot.org/proteomes?facets=proteome_type:1&query=(organism_id:10090)).
The function identifies the position of the modified amino acid(s) in a
master protein, and uses that information as a midpoint to calculate the
position range of amino acids in a motif. The blast+ suite is then used
to search a proteome and extract the motifs. To do this, a user defined
proteome is converted to a blast searchable database (using
makeblastdb). The master protein accession is used to search the
database (using blastdbcmd), and if present the motif determined by the
range calculated above is extracted and appended to the xlsx table.

If multiple master proteins or multiple modified amino acids are
identified for a given row of input data, this function will separate
that data to search a single master protein / modification at a time
resulting in a greater number of rows in the data set.

### Function Contents

```         
run_motif_extraction.R
build_blastdb.R
formatter.R
data_pull_and_fill.R
run_blastdbcmd.R
```

### Parameters

This program has 5 parameters. 1. The path to the data file - An .xlsx
peptidel isoforms output file from Proteome Discover (is this correct?)
that must includes the following columns information: \|
Modifications_in_Master_Proteins \| \| ---- \| \| P49312 1xPhospho
[S308(99.1)] \| \| A0A668KLC6 2xPhospho [T1811(100); S1815(99.2)] \| \|
etc... \| **SXXX** indicates the modified amino acid (in this case the
phosphorylated amino acid) 2. The user determined path for the output
file 3. The path to ncbi-blast-2.14.0+/bin - download the [blast+
executables](https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/)
4. The path to Fasta reference file for desired organism [e.g. mouse
proteome
download](https://rest.uniprot.org/uniprotkb/stream?download=true&format=fasta&query=%28mouse%29%20AND%20%28model_organism%3A10090%29))\
5. The number of amino acids on either side of the modified amino acid
to extract from the Master protein (default is num=7)

### The output

-   an .csv file with additional columns
    -   accession
    -   start
    -   end
    -   range
    -   extracted_sequence

### Example usage

```{r run_motif_extraction()}
#path to the protein sequence for the reference database
ref_protein <- "/Users/eguswa/Documents/Open_UVM_Projects/Proteomics_core/Mouse_ref/protein.faa"

#path to bin in blast suit
ncbi_bin <- "/Applications/ncbi-blast-2.14.0+/bin"

#input data sheet
input_path <- "/Users/eguswa/Documents/Open_UVM_Projects/Proteomics_core/Peptide_extraction/May222023_Eclipse_2022_138_unnormalized_peptideIsoforms_Bioinformatics.xlsx"

#output path
out_dir_path <- "/Users/eguswa/Documents/Open_UVM_Projects/Proteomics_core/Peptide_extraction/May222023_Eclipse_2022_138_with_extracted_sequence.csv"

run_motif_extraction(input_path, output_path, ncbi_bin, ref_protein)
```
