<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>UVM_Proteomics • UVMProteomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="UVM_Proteomics">
<meta property="og:description" content="This package is designed for the UVM Proteomics Core to finds the 
    7 residues before and after given accessions and phosphosites of interest. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">UVMProteomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="figure">
<img src="/logo_landscape.png" alt=""><p class="caption">VBRN Logo</p>
</div>
<div class="section level1">
<div class="page-header"><h1 id="uvm-proteomics">
<a href="https://vbrn.org/proteomics/" class="external-link">UVM Proteomics</a><a class="anchor" aria-label="anchor" href="#uvm-proteomics"></a>
</h1></div>
<div class="section level3">
<h3 id="authors">Authors<a class="anchor" aria-label="anchor" href="#authors"></a>
</h3>
<p><a href="https://scholar.google.com/citations?user=uGHWHbgAAAAJ&amp;hl=en&amp;oi=ao" class="external-link">Emily Curd</a> and James (Jax) Lubkowitz</p>
</div>
<div class="section level3">
<h3 id="affiliations">Affiliations<a class="anchor" aria-label="anchor" href="#affiliations"></a>
</h3>
<p><a href="https://vbrn.org" class="external-link">Vermont Biomedical Research Network</a>, <a href="https://www.med.uvm.edu/vigr/home" class="external-link">Vermont Integrated Genomics Resource</a>, <a href="https://www.uvm.edu" class="external-link">University of Vermont</a>, <a href="https://www.stlawu.edu" class="external-link">St. Lawrence University</a></p>
</div>
<div class="section level3">
<h3 id="funding">Funding<a class="anchor" aria-label="anchor" href="#funding"></a>
</h3>
<p>Research reported in this repository was supported by an Institutional Development Award (IDeA) from the National Institute of General Medical Sciences of the National Institutes of Health under grant number P20GM103449. Its contents are solely the responsibility of the authors and do not necessarily represent the official views of NIGMS or NIH.</p>
<p>This is a R package for the <a href="https://vbrn.org/proteomics/" class="external-link">UVM - VBRN Proteomics Core</a>. It includes functions for three main types of analyses: Post Translational Modification, Spectral Counting, and Protein Expression.</p>
<p><strong><em>Currently Supports or In Development</em></strong></p>
<p>Post Translational Modification: - Motif Analysis</p>
<p>Spectral Counting: - Differential protein expression (DEqMS)</p>
<p>Protein Expression: - Pathway Analysis - functional classification</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Package can be installed from github via</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"UVM_Proteomics"</span>,username<span class="op">=</span><span class="st">"jaxlub"</span><span class="op">)</span></span></code></pre></div>
<p>Another download option is to download a zip file from github, unzip it and then run</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="va">source</span> <span class="op">&lt;-</span> <span class="fu">devtools</span><span class="fu">:::</span><span class="fu">source_pkg</span><span class="op">(</span><span class="st">"path/to/package"</span><span class="op">)</span></span>
<span><span class="fu">install</span><span class="op">(</span><span class="va">source</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="post-translational-modification">Post Translational Modification<a class="anchor" aria-label="anchor" href="#post-translational-modification"></a>
</h1>
<div class="section level2">
<h2 id="motif-extraction">Motif Extraction<a class="anchor" aria-label="anchor" href="#motif-extraction"></a>
</h2>
<p>This function extracts predetermined protein motifs from a master protein in a user specified proteome (e.g. <a href="https://www.uniprot.org/proteomes?facets=proteome_type:1&amp;query=(organism_id:10090)" class="external-link">uniProt mouse proteome</a>. The function identifies the position of the modified amino acid(s) in a master protein, and uses that information as a midpoint to calculate the position range of amino acids in a motif. The blast+ suite is then used to search a proteome and extract the motifs. To do this, a user defined proteome is converted to a blast searchable database (using makeblastdb). The master protein accession is used to search the database (using blastdbcmd), and if present the motif determined by the range calculated above is extracted and appended to the xlsx table.</p>
<p>If multiple master proteins or multiple modified amino acids are identified for a given row of input data, this function will separate that data to search a single master protein / modification at a time resulting in a greater number of rows in the data set.</p>
<div class="section level3">
<h3 id="function-contents">Function Contents<a class="anchor" aria-label="anchor" href="#function-contents"></a>
</h3>
<pre><code><span><span class="va">run_motif_extraction.R</span></span>
<span><span class="va">build_blastdb.R</span></span>
<span><span class="va">formatter.R</span></span>
<span><span class="va">data_pull_and_fill.R</span></span>
<span><span class="va">run_blastdbcmd.R</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="parameters">Parameters<a class="anchor" aria-label="anchor" href="#parameters"></a>
</h3>
<p>This program has 5 parameters. 1. The path to the data file - An .xlsx peptidel isoforms output file from Proteome Discover (is this correct?) that must includes the following columns information: | Modifications_in_Master_Proteins | | —- | | P49312 1xPhospho [S308(99.1)] | | A0A668KLC6 2xPhospho [T1811(100); S1815(99.2)] | | etc… | <strong>SXXX</strong> indicates the modified amino acid (in this case the phosphorylated amino acid) 2. The user determined path for the output file 3. The path to ncbi-blast-2.14.0+/bin - download the <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/" class="external-link">blast+ executables</a> 4. The path to Fasta reference file for desired organism <a href="https://rest.uniprot.org/uniprotkb/stream?download=true&amp;format=fasta&amp;query=%28mouse%29%20AND%20%28model_organism%3A10090%29" class="external-link">e.g. mouse proteome download</a>)<br>
5. The number of amino acids on either side of the modified amino acid to extract from the Master protein (default is num=7)</p>
</div>
<div class="section level3">
<h3 id="the-output">The output<a class="anchor" aria-label="anchor" href="#the-output"></a>
</h3>
<ul>
<li>an .csv file with additional columns
<ul>
<li>accession</li>
<li>start</li>
<li>end</li>
<li>range</li>
<li>extracted_sequence</li>
</ul>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="example-usage">Example usage<a class="anchor" aria-label="anchor" href="#example-usage"></a>
</h3>
<pre><code><span><span class="co">#path to the protein sequence for the reference database</span></span>
<span><span class="va">ref_protein</span> <span class="op">&lt;-</span> <span class="st">"/Users/eguswa/Documents/Open_UVM_Projects/Proteomics_core/Mouse_ref/protein.faa"</span></span>
<span></span>
<span><span class="co">#path to bin in blast suit</span></span>
<span><span class="va">ncbi_bin</span> <span class="op">&lt;-</span> <span class="st">"/Applications/ncbi-blast-2.14.0+/bin"</span></span>
<span></span>
<span><span class="co">#input data sheet</span></span>
<span><span class="va">input_path</span> <span class="op">&lt;-</span> <span class="st">"/Users/eguswa/Documents/Open_UVM_Projects/Proteomics_core/Peptide_extraction/May222023_Eclipse_2022_138_unnormalized_peptideIsoforms_Bioinformatics.xlsx"</span></span>
<span></span>
<span><span class="co">#output path</span></span>
<span><span class="va">out_dir_path</span> <span class="op">&lt;-</span> <span class="st">"/Users/eguswa/Documents/Open_UVM_Projects/Proteomics_core/Peptide_extraction/May222023_Eclipse_2022_138_with_extracted_sequence.csv"</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="reference/run_motif_extraction.html">run_motif_extraction</a></span><span class="op">(</span><span class="va">input_path</span>, <span class="va">output_path</span>, <span class="va">ncbi_bin</span>, <span class="va">ref_protein</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>FALSE</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing UVMProteomics</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Emily Guswa <br><small class="roles"> Maintainer </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Emily Guswa.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
